### DISCUSSION TITLE
/lib64/libm.so.6: version `GLIBC_2.23' not found in RedHat

### DISCUSSION THREAD

**Discussion by yxmm-wxe** [duckdb/duckdb#2676] [Q&A] [answered]
When I use DuckDB via duckdb_jdbc 0.3.1 on Ubuntu 18.04.6 LTS, it works.

However, when I use DuckDB via duckdb_jdbc 0.3.1 on Red Hat Enterprise Linux Server release 7.6 (Maipo), it will throw the following issue.

```
java.lang.UnsatisfiedLinkError: /tmp/libduckdb_java9113316470680153190.so: /lib64/libm.so.6: version `GLIBC_2.23' not found (required by /tmp/libduckdb_java9113316470680153190.so)
```

I'd like to know if my environment is missing some packages or DuckDB issue?

**Comment by hannes** [comment, marked as answer]
The GLIBC on the Redhat is too old. Can you build from source there?

**Reply by yxmm-wxe** [reply]
Is there any documentation describing the DuckDB dependencies and their versions? I am worried that there are other dependencies missing after solving this.

**Reply by hannes** [reply]
No documentation as far as I'm aware. To build from source, you will need GCC 4.8 or newer, cmake 2.8 or newer. I suggest you give it a try.

**Reply by yxmm-wxe** [reply]
I found the same issue on CentOS 7.

I tried to build from source to get duckdb_jdbc that depends on low version GLIBC on CentOS 7.

```bash
BUILD_JDBC=1 make
```

I got the following products via the above command.

```bash
total 29660
drwxrwxr-x. 5 xavier xavier 164 Dec 1 10:48 CMakeFiles
-rw-rw-r--. 1 xavier xavier 1251 Dec 1 10:48 cmake_install.cmake
-rw-rw-r--. 1 xavier xavier 7019290 Dec 1 11:02 duckdb_jdbc.jar
-rwxrwxr-x. 1 xavier xavier 23332248 Dec 1 11:02 libduckdb_java.so_linux_amd64
-rw-rw-r--. 1 xavier xavier 9413 Dec 1 10:48 Makefile
```

The size of duckdb_jdbc release 0.3.1 is about 29M.
The size of my own built duckdb_jdbc jar is about 7M. But I found the size of the result directory is about 29M.
Compare to duckdb_jdbc release 0.3.1, my own built jar is much smaller than the released one.

I wonder if the build is missing some parameters or other issues. Are there instructions for building duckdb_jdbc?

Another question, Is there any plan to support the .NET driver.

**Reply by hannes** [reply]
The release JAR is bigger since it contains binaries for multiple platforms (Windows, Linux, OSX). So its very expected that yours is smaller. Does it work?

**Reply by yxmm-wxe** [reply]
Thanks for your explanation. I will try to see if it works. How can I build from sources to get duckdb_jdbc for multiple platforms? I need the duckdb_jdbc that depends on low version GLIBC for multiple platforms (Windows, Linux, OSX).

**Reply by hannes** [reply]
Could you perhaps tell me what you need this for?

**Reply by yxmm-wxe** [reply]
We need an in-memory analytics engine to analyze user data. And we also need to support multiple platforms.
DuckDB is an excellent in-process database, we use it as an in-memory analytics engine.

**Reply by hannes** [reply]
And who is "We"?

**Reply by yxmm-wxe** [reply]
I tried to use my own built duckdb_jdbc that depends on GLIBC_2.17 on CentOS 7. It works! Thanks for your help.

We are the development team of [Wyn](https://wyn.grapecity.com/).

Could you help get duckdb_jdbc for multiple platforms?

**Reply by hannes** [reply]
This starts to look a lot like free commercial support, could you please get in touch at hannes@duckdblabs.com to discuss options?

**Reply by yxmm-wxe** [reply]
OK, I will try to get in touch.
